fix: add flatpak ci

rohitpid opened the pull request:
## Contributor Checklist:

* [ ] This is a user-visible change and I have created a file in the `doc/newsfragments` directory (and made sure to read the `README.md` in that directory)
* [ ] This is not a user-visible change

sithlord48 committed:
build: Require Cmake 3.21

sithlord48 committed:
ci: add CI to generate a flatpak and build qt6 GUI

* add python build deps for libei 1.3.0
* build libportal 0.8.1 with qt6 backend
* build flatpak with org.kde.Sdk and Platform
* build the qt6 GUI

KoljaFrahm committed:
ci: make libei required for Fedora and SUSE

Makes libei required so the dnf work with wayland.
Also unifies the dnf build between Fedora and SUSE, so that mostly pkgconfig are required.
Removes unnecessary dependencies.

Co-authored-by: Neal Gompa <ngompa@fedoraproject.org>

GeorgesStavracas commented:
Seems like the Gamescope vulkan layer for Flatpak builds libei too: https://github.com/flathub/org.freedesktop.Platform.VulkanLayer.gamescope/blob/branch/23.08/modules/libei.yml

You can maybe try that

rohitpid commented on dist/flatpak/input-leap-flatpak:
```diff
@@ -9,7 +9,6 @@ fi
 if [[ -z "$XDG_RUNTIME_DIR" ]]; then
     XDG_RUNTIME_DIR="/tmp"
 fi
-export LIBEI_SOCKET=$XDG_RUNTIME_DIR/eis-0
```
@GeorgesStavracas Do you think adding this line would help? I don't think it impacts the GUI at all so maybe only for the helper script.

ofourdan commented:
Not sure whether this is intentional or not, but it looks you've created the PR against your own repo, instead of [input-input/input-leap-leap](https://github.com/input-leap/input-leap/) so it's very unlikely upstream input-leap will see or review it in there.

whot commented:
Mini-feedback after 2 min of testing: works between Fedora 40 and Fedora 40, both are dual screen setups and that worked too. Didn't test much more than mouse movements and a few key strokes but it did work, so - nice job!

What's missing is a `.desktop` file though and a few other things like license information (appinfo? I forgot where all that goes). I had to start it via `flatpak run` from the commandline because GNOME software doesn't even show it as installed - presumably that's the missing `.desktop` file.

sithlord48 committed:
Add dpiAwareness setting to Window manifests

This fixes per-desktop scale on Windows.

Co-authored-by: Povilas Kanapickas <povilas@radix.lt>

sithlord48 committed:
build: Remove un needed INPUTLEAP_BUILD_INSTALLER option

sithlord48 committed:
build: Update AppData Deployment

sithlord48 committed:
fix: gate leave() with canLeave(), allowing cleanup

Koston-0xDEADBEEF committed:
gui: Add missing signal.h include

Fixes build error such as:

src/gui/src/MainWindow.cpp:843:9: error: use of undeclared identifier
'kill'

Note that kill is from Posix, not C++, so csignal cannot be used.

Koston-0xDEADBEEF committed:
ci: Add FreeBSD CI build

sithlord48 committed:
gui: Show window and file menu on mac os

p12tic committed:
Merge pull request #1970 from sithlord48/cmake_rm_unneeded_option

build: Remove un needed INPUTLEAP_BUILD_INSTALLER option

shymega committed:
Merge pull request #1971 from sithlord48/update_metadata

build: Update AppData Deployment

p12tic committed:
Merge pull request #1975 from sithlord48/macosmenu

fix: show window and file menu on mac os

p12tic committed:
Merge pull request #1973 from Koston-0xDEADBEEF/fbsd-build

Fix build on FreeBSD

p12tic committed:
github: Clarify criteria for inclusion of newsfragment

User-visible is not entirely clear because it may be interpreted
literally (i.e. must change something that user sees). The intended
criteria for newsfragment is that any changes affecting end users should
have a release note.

Changes that don't affect end users, such as refactors or build system
changes do not need release notes.

p12tic committed:
Merge pull request #1974 from Koston-0xDEADBEEF/fbsd-github-action

FreeBSD CI build

p12tic committed:
Merge pull request #1968 from sithlord48/winscaleFix

fix: Scale per display on windows

p12tic committed:
Merge pull request #1978 from p12tic/pr-template-clarify-newsfragment

github: Clarify criteria for inclusion of newsfragment

sithlord48 committed:
build: Remove the dependency on curl since it was unused

sithlord48 committed:
build: remove stale build scripts

shymega committed:
Merge pull request #1980 from sithlord48/depRemoval

build: remove unused curl dependency

shymega committed:
Merge pull request #1972 from sithlord48/keystuckfix

fix: stuck keys on client

shymega committed:
Merge pull request #1981 from sithlord48/rmBuildscripts

build: remove stale build scripts

shymega committed:
Merge pull request #1804 from sithlord48/requireCmake321

Require Cmake 3.21

sithlord48 committed:
ci: use ubuntu 22.04 version of cmake

p12tic committed:
Revert "build: remove stale build scripts"

p12tic committed:
Merge pull request #1983 from sithlord48/cmakefromubuntu

ci: use ubuntu 22.04 version of cmake

p12tic committed:
Merge pull request #1985 from input-leap/revert-1981-rmBuildscripts

Revert "build: remove stale build scripts"

p12tic committed:
Merge pull request #1954 from KoljaFrahm/require-libei

Require libei
